### ISSUE TITLE
Test fails

### ISSUE THREAD

**Issue by wanderley** [compnerd/swift-build-examples#7]
I did the following

```
cd ~/git/swift-cmake-demo/
rm -rf build
mkdir build
cd build
cmake -S ../ -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=YES
ninja
ninja test
```

and I got

```
dyld: Library not loaded: Source/HelloWorldCore/libHelloWorldCore.dylib
  Referenced from: /Users/wander/git/swift-cmake-demo/build/Tests/TestHelloWorldCore/TestHelloWorldCore
  Reason: image not found
Abort trap: 6
```

By the way, I can run `build/Source/HelloWorld/HelloWorld` without problems.

**Comment by compnerd** [comment]
Hmm, we should figure out how to ensure that `LC_RPATH` is setup properly and reset during installation.

**Comment by compnerd** [comment]
This should be fixed in 3.16 I believe.
